From eebbc01d280b81b3d99948eeef2008f9f10c04c9 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@opensdr.com>
Date: Wed, 28 Oct 2015 10:45:27 -0700
Subject: [PATCH 14/14] If we are cross compiling, do not test for build
 machine distro.

UHD would use the result of the REDHAT test to decide if UHD installed
in lib64. So if you did a build for a 64 bit machine on a FEDORA box with
OE, it decided to install uhd in /usr/lib64. This is not always the right thing
to do.

Eventually, we will need to deal with OE builds that use lib64, but we can figure
that out when we hit the problem.

Signed-off-by: Philip Balister <philip@opensdr.com>
---
 host/cmake/Modules/UHDPackage.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/host/cmake/Modules/UHDPackage.cmake b/host/cmake/Modules/UHDPackage.cmake
index 6409358..08c2a6c 100644
--- a/host/cmake/Modules/UHDPackage.cmake
+++ b/host/cmake/Modules/UHDPackage.cmake
@@ -27,11 +27,11 @@ IF(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     SET(LINUX TRUE)
 ENDIF()
 
-IF(LINUX AND EXISTS "/etc/debian_version")
+IF(NOT CMAKE_CROSSCOMPILING AND LINUX AND EXISTS "/etc/debian_version")
     SET(DEBIAN TRUE)
 ENDIF()
 
-IF(LINUX AND EXISTS "/etc/redhat-release")
+IF(NOT CMAKE_CROSSCOMPILING AND LINUX AND EXISTS "/etc/redhat-release")
     SET(REDHAT TRUE)
 ENDIF()
 
-- 
2.1.4

